lib/vector-core/src/stream/driver.rs

```diff
@@ -190,11 +193,26 @@ where
                         byte_size: cbs.1,
                         output: None,
                     });
+
+                // This condition occurs specifically when the `HttpBatchService::call()` is called *within* the `Service::call()`
+                } else if response.event_status() == EventStatus::Rejected {
+                    Self::emit_call_error(None, request_id, metadata.event_count());
+                    finalizers.update_status(EventStatus::Rejected);
                 }
             }
         };
         drop(finalizers); // suppress "argument not consumed" warning
     }
+
+    /// Emit the `Error` and `EventsDropped` internal events.
+    /// This scenario occurs after retries have been attempted.
+    fn emit_call_error(error: Option<Svc::Error>, request_id: usize, count: usize) {
+        emit(CallError {
+            error,
+            request_id,
+            count: count as u64,
```

Comment:
This actually reminds me of something I noticed from one of Bruce's recent PRs that I thought of modelling the `ComponentEventsDropped` after.

Instead of having the `ComponentEventsDropped.count` be a u64, just have it be a usize.
And do the cast at the very end there.

(That is why this one is a u64).

I can change that here.

Reply:
This also made me realize there was a nasty merge issue in this area which I'll fix.

Reply:
Done per https://github.com/vectordotdev/vector/pull/14836/commits/dae96939af37f21f9d662167a90224daaa0168f3